pkg/kv/kvserver/replica_destroy.go

```diff
@@ -88,23 +88,10 @@ func (r *Replica) destroyRaftMuLocked(ctx context.Context, nextReplicaID roachpb
 	ms := r.GetMVCCStats()
 	batch := r.store.TODOEngine().NewWriteBatch()
 	defer batch.Close()
-	desc := r.Desc()
-	inited := desc.IsInitialized()
-
-	opts := kvstorage.ClearRangeDataOptions{
-		ClearReplicatedBySpan: desc.RSpan(), // zero if !inited
-		// TODO(tbg): if it's uninitialized, we might as well clear
-		// the replicated state because there isn't any. This seems
-		// like it would be simpler, but needs a code audit to ensure
-		// callers don't call this in in-between states where the above
-		// assumption doesn't hold.
-		ClearReplicatedByRangeID:   inited,
```

Comment:
NB: after this PR, `ClearReplicatedByRangeID` is effectively set to true here, i.e. the TODO is done. I will double-check tomorrow, but I don't think there an in-between state that this TODO talks about since all replica storage updates are atomic, and we have extra assurance due to `raftMu`. For uninitialized replicas, this span will be empty, and a non-empty one would be a bug. So clearing it unconditionally makes the code more resilient.